crates/graphql/src/schema/query_root.rs

```diff
@@ -308,16 +312,42 @@ impl QueryRoot {
             && creators.is_none()
             && auction_houses.is_none()
             && offerers.is_none()
+            && term.is_none()
         {
             return Err(FieldError::new(
                 "No filter provided! Please provide at least one of the filters",
-                graphql_value!({ "Filters": "owners: Vec<PublicKey>, creators: Vec<PublicKey>, offerers: Vec<PublicKey>, auction_houses: Vec<PublicKey>" }),
+                graphql_value!({ "Filters": "owners: Vec<PublicKey>, creators: Vec<PublicKey>, offerers: Vec<PublicKey>, auction_houses: Vec<PublicKey>, term: String" }),
             ));
         }
 
         let conn = context.shared.db.get().context("failed to connect to db")?;
 
+        let addresses = match term {
+            Some(term) => {
+                let search = &context.shared.search;
+                let search_result = search
+                    .index("metadatas")
+                    .search()
+                    .with_query(&term)
+                    .with_offset(offset.try_into()?)
+                    .with_limit(limit.try_into()?)
+                    .execute::<Value>()
+                    .await
+                    .context("failed to load search result for metadata json")?
+                    .hits;
+
+                Some(
+                    search_result
+                        .into_iter()
+                        .map(|r| MetadataJson::from(r.result).address)
+                        .collect(),
+                )
+            },
+            None => None,
+        };
```

Comment:
You can use a map on the optional term to run the transform when the term has some value.

```suggestion
        let search_addresses = term.map(|term| {
                 let search = &context.shared.search;
                let search_result = search
                    .index("metadatas")
                    .search()
                    .with_query(&term)
                    .with_offset(offset.try_into()?)
                    .with_limit(limit.try_into()?)
                    .execute::<Value>()
                    .await
                    .context("failed to load search result for metadata json")?
                    .hits;
        })
```

Reply:
Do you know an elegant way to use `Option::map` with an async operation? [My googling](https://users.rust-lang.org/t/how-to-run-async-code-within-option-result-function-chain/64053) makes it seem like you have to extract the operation to a separate function and then use that function. In that case, what I currently have is clearer/cleaner.

Reply:
Ah the async nvm then.